datafusion/physical-plan/src/joins/sort_merge_join.rs

```diff
@@ -370,8 +370,15 @@ impl DisplayAs for SortMergeJoinExec {
                 )
             }
             DisplayFormatType::TreeRender => {
-                // TODO: collect info
-                write!(f, "")
+                let on = self
+                    .on
+                    .iter()
+                    .map(|(c1, c2)| format!("({} = {})", c1, c2))
+                    .collect::<Vec<String>>()
+                    .join(", ");
+
+                writeln!(f, "join_type={:?}", self.join_type)?;
```

Comment:
I recommend the same things here: 
https://github.com/apache/datafusion/pull/15079#discussion_r1986065338